### PR TITLE
[SW-531] Fix error when opening the "Terms of use" from Index page and trying to go back

### PR DESCRIPTION
**Pull request by pstachel**

### CODE REVIEW

File: src/popup/components/Header.vue [hunk 1]
```diff
@@ -107,6 +107,9 @@ export default defineComponent({
     );
 
     function back() {
+      if (root.$route.meta?.directBackRoute) {
```
Comment (peronczyk) [inline src/popup/components/Header.vue:110]: I suggest to move the line 117 before this test and pick the `meta` the same way as the `fullPath` then use the `meta` here and in line 113.

**Review by peronczyk — APPROVED** [review]

File: src/popup/components/Header.vue (hunk 1 above)
Comment (pstachel) [reply src/popup/components/Header.vue:110]: good idea

File: src/popup/components/Header.vue [hunk 2]
```diff
@@ -107,14 +107,18 @@ export default defineComponent({
     );
 
     function back() {
-      if (root.$route.meta?.backRoute) {
+      const { fullPath, meta } = root.$route;
+
+      if (meta?.directBackRoute) {
+        return root.$router.go(-1);
+      }
+      if (meta?.backRoute) {
```
Comment (subhod-i) [inline src/popup/components/Header.vue:115]: `meta.backRoute` is accessed twice. Could be moved to a variable.

File: src/popup/components/Header.vue [hunk 3]
```diff
@@ -107,14 +107,18 @@ export default defineComponent({
...
         // TODO: rewrite back button logic in more unified way
-        return root.$router.push(root.$route.meta?.backRoute);
+        return root.$router.push(meta?.backRoute);
       }
-      let { fullPath } = root.$route;
-      fullPath = fullPath.endsWith('/') ? fullPath.slice(0, -1) : fullPath;
+      const path = fullPath.endsWith('/') ? fullPath.slice(0, -1) : fullPath;
       return root.$router.push(
-        fullPath.substr(0, fullPath.lastIndexOf('/')) || { name: currentHomeRouteName.value },
+        path.substr(0, path.lastIndexOf('/')) || { name: currentHomeRouteName.value },
```
Comment (subhod-i) [inline src/popup/components/Header.vue:121]: Ternary and '||" operators are used sequentially. Decreases the code readability.

Could be rewritten to
```js
    const path = fullPath.endsWith('/') ? fullPath.slice(0, -1) : fullPath;
    const lastSlashIndex = path.lastIndexOf('/');
    if (lastSlashIndex !== -1) {
      path = path.substr(0, lastSlashIndex);
    } else {
      path = { name: currentHomeRouteName };
    }
    return root.$router.push(path);
```

**Pull request merged**